extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php

```diff
@@ -14,13 +14,89 @@ public function __construct() {
 	}
 
 	/**
+	 * Sends the weekly digest to all users in the global_watchlist table
+	 */
+	public function sendWeeklyDigest() {
+		foreach ( $this->getUserIDs() as $userID ) {
+			$this->sendDigestToUser( $userID );
+			$this->clearWatchLists( $userID );
+		}
+	}
+
+	/**
+	 * Return all users in the global_watchlist table
+	 * @return array
+	 */
+	private function getUserIDs() {
+		$db = wfGetDB( DB_SLAVE, [], \F::app()->wg->ExternalDatawareDB );
+		$userIDs = ( new WikiaSQL() )
+			->SELECT()->DISTINCT( GlobalWatchlistTable::COLUMN_USER_ID )
+			->FROM( GlobalWatchlistTable::TABLE_NAME )
+			->runLoop( $db, function ( &$userIDs, $row ) {
+				$userIDs[] = $row->gwa_user_id;
+			} );
+
+		return $userIDs;
+	}
+
+	/**
+	 * Clear the global_watchlist and local watchlists for a given user.
+	 * This is done after we send them the weekly digest which effectively
+	 * means they have "seen" all the watched pages and will receive notifications
+	 * for new edits.
+	 * @param $userID
```

Comment:
Could use a type here.  I'll add this same suggestion to the other lines rather than name them all here because I'm lazy.